### ISSUE TITLE
reth 1.1.0 crashed due to "cannot convert a slice of length 30 to FixedBytes<32>" in consensus engine

### ISSUE THREAD

**Issue by TPXP** [paradigmxyz/reth#12934]
### Describe the bug

Hello,

On Nov 24th, 2024 05:32:44 UTC, our 2 instances of reth 1.1.0 crashed with the same error.
```
thread 'Tree Task' panicked at crates/trie/trie/src/witness.rs:352:40:
cannot convert a slice of length 30 to FixedBytes<32>
stack backtrace:
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
2024-11-28T05:32:41.200025Z ERROR Fatal error
2024-11-28T05:32:41.200055Z ERROR Fatal error in consensus engine
2024-11-28T05:32:41.200221Z ERROR shutting down due to error
Error: Fatal error in consensus engine

Location:
    /project/crates/node/builder/src/launch/engine.rs:367:43
```

The two instances are paired with separate Lighthouse 5.3.0 instances. We find the fact that they crashed at the very same time (and with the same error) quite suspicious. We're not quite sure where the fault lies (could it be a faulty payload from Lighthouse?) but nothing special appears in Lighthouse logs.
```
Nov 28 05:32:24.930 INFO New block received                      root: 0x0c3eedab29d6da5ff63f7b835e8abf43ae64f74c5368af5482c96ea5d2f0524b, slot: 10495660
Nov 28 05:32:29.000 INFO Synced                                  slot: 10495660, block: 0x0c3e…524b, epoch: 327989, finalized_epoch: 327987, finalized_root: 0x7545…a17a, exec_hash: 0x6bda…d036 (verified), peers: 104, service: slot_notifier
Nov 28 05:32:36.548 INFO New block received                      root: 0x879b8daa3c355c6407035fbd47dbafefe8e72e23e84d79ee99ab029b0d7df276, slot: 10495661
Nov 28 05:32:41.000 INFO Synced                                  slot: 10495661, block:    …  empty, epoch: 327989, finalized_epoch: 327987, finalized_root: 0x7545…a17a, exec_hash: 0x6bda…d036 (verified), peers: 102, service: slot_notifier
Nov 28 05:32:41.308 WARN Execution engine call failed            error: ServerMessage { code: -32603, message: "Server error" }, service: exec
Nov 28 05:32:41.308 WARN Error whilst processing payload status  error: Api { error: ServerMessage { code: -32603, message: "Server error" } }, service: exec
Nov 28 05:32:41.309 ERRO Error during execution engine upcheck   error: HttpClient(url: http://mainnet-reth-archive-rpc-0-0.mainnet-reth-archive-rpc-0:8551/, kind: request, detail: connection error: Connection reset by peer (os error 104)), service: exec
Nov 28 05:32:42.499 WARN Execution engine call failed            error: HttpClient(url: http://mainnet-reth-archive-rpc-0-0.mainnet-reth-archive-rpc-0:8551/, kind: request, detail: error trying to connect: tcp connect error: Connection refused (os error 111)), service: exec
Nov 28 05:32:42.500 ERRO Error during execution engine upcheck   error: HttpClient(url: http://mainnet-reth-archive-rpc-0-0.mainnet-reth-archive-rpc-0:8551/, kind: request, detail: error trying to connect: tcp connect error: Connection refused (os error 111)), service: exec
Nov 28 05:32:42.587 ERRO Error fetching block for peer           error: EngineGetCapabilititesFailed(ApiError(HttpClient(url: http://mainnet-reth-archive-rpc-0-0.mainnet-reth-archive-rpc-0:8551/, kind: request, detail: error trying to connect: tcp connect error: Connection refused (os error 111)))), block_root: 0x7d01aa8
fdb5ef3916b1e1c0e188e7169c5410b01c27cf98891eb3648c4860f17
```

I don't know if the bug was fixed in a recent reth release, but in case it wasn't it would be awesome if this (rare) case could be handled and prevent reth from crashing completely. 🙏 Having our 2 reth instances crashing at the same time defeats any high availability strategy we could put in place on our reth RPC endpoint.

### Steps to reproduce

Not sure, after restarting reth resumed and stayed synced with the chain.

### Platform(s)

Linux (x86)

### What version/commit are you on?

reth 1.1.0 - using docker image `ghcr.io/paradigmxyz/reth:v1.1.0`

```
reth Version: 1.1.0
Commit SHA: 1ba631ba9581973e7c6cadeea92cfe1802aceb4a
Build Timestamp: 2024-10-10T18:21:07.285015202Z
Build Features: asm_keccak,jemalloc
Build Profile: maxperf
```

### What database version are you on?

We've configured a custom data path with `--datadir` (`error: unexpected argument '--datadir' found`). It seems `reth db version` does not support this flag so all we get is
```
2024-11-28T06:39:14.234940Z  INFO Initialized tracing, debug log directory: /root/.cache/reth/logs/mainnet
Error: Datadir does not exist: "/root/.local/share/reth/mainnet"

Location:
    /project/crates/cli/commands/src/db/mod.rs:76:9
```

### Which chain / network are you on?

mainnet

### What type of node are you running?

Archive (default)

### What prune config do you use, if any?

We don't have a `reth.toml` file. Here's our full reth command instead
```
reth node \
  --datadir=/data \
  --port=30303 \
  --discovery.port=30303 \
  --http \
  --http.api=eth,net,trace,txpool,web3,rpc,debug \
  --http.addr=0.0.0.0 \
  --http.corsdomain=* \
  --http.port=8545 \
  --ws \
  --ws.api=eth,net,trace,txpool,web3,rpc,debug \
  --ws.addr=0.0.0.0 \
  --ws.port=8545 \
  --ws.origins=* \
  --authrpc.addr=0.0.0.0 \
  --nat=publicip \
  --authrpc.jwtsecret=/jwt-secret/jwt \
  --metrics=0.0.0.0:6060 \
  --chain=mainnet \
  --rpc-max-connections=10000
```

### Code of Conduct

- [x] I agree to follow the Code of Conduct

**Comment by rkrasiuk** [comment]
@TPXP it's not computing bad blocks but rather panicking on trying to gather additional data after computing bad blocks. possibly related to https://github.com/paradigmxyz/reth/issues/12934

**Comment by TPXP** [comment]
@rkrasiuk Thanks for the investigation. I think reth should gracefully recover when getting a bad block instead 🙏 

Also, the issue you referenced in your comment is this issue, you probably meant something else 😉

**Comment by aimxhaisse** [comment]
For the record, likely related to https://x.com/preston_vanloon/status/1862031578454704612

**Comment by rkrasiuk** [comment]
@TPXP oops, fixed

**Comment by rkrasiuk** [comment]
@aimxhaisse you are correct, that's the reason for the invalid block hook to kick in

@aimxhaisse @TPXP providing more context: we have an invalid block hook that kicks upon encountering a bad block. usually, it's a helpful tool for us to debug outstanding issues ahead of the release. unfortunately, in this case, the hook itself resulted in a panic, putting out a PR soon to address this

**Comment by rkrasiuk** [comment]
@aimxhaisse @TPXP actually, upon the second review, this specific issue you encountered has been fixed in `v1.1.2` via https://github.com/paradigmxyz/reth/commit/12cab204b50a824bcfb55f6b6d46f2e9cda34c31

**Comment by TPXP** [comment]
Oh, that's great. Thanks a lot for the quick analysis, we'll upgrade our reth instances. 👍
